### ISSUE TITLE
Gateio not returning withdrawals for Portfolio Margin Account

### ISSUE THREAD

**Issue by dwiltech** [ccxt/ccxt#18222]
### Operating System

Windows 10

### Programming Languages

python

### CCXT Version

3.1.6

### Description

When fetching withdrawals from gateio exchange the return does not include any withdrawals for a Portfolio Margin Account. I can see withdrawals which were made within the last 7 days on the exchange UI. I also get the same result if I specify the currency (optional).  I do not think it involves the optional time range parameters since withdrawals exist on the exchange which were made a few days of making the fetch and that is within the timeframe of the 'from' default of 7 days. I can get the balance for gateio using the parameter: 'marginMode':'cross', (thank you ttodua), but something like that does not seem to be an option for fetching withdrawals.

fetch Request: gateio GET https://api.gateio.ws/api/v4/wallet/withdrawals RequestHeaders: {'KEY': 'nnnnd3ced26a24042af275nnnn', 'Timestamp': '1686737456', 'SIGN': 'nnnnnnn0b4696afbe966555630f35f9e9f126db9c2c900b7c4d6139edfc9b527ebcecd8e7299ef9902d6d0ab5b58c6ee215afba81e576284c9nnnnnnnnnn', 'Content-Type': 'application/json', 'User-Agent': 'python-requests/2.28.2', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive', 'X-Gate-Channel-Id': 'ccxt'} RequestBody: None

fetch Response: gateio GET https://api.gateio.ws/api/v4/wallet/withdrawals 200 ResponseHeaders: {'Date': 'Wed, 14 Jun 2023 10:10:57 GMT', 'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'Server': 'openresty', 'X-Frame-Options': 'SAMEORIGIN', 'X-Powered-By': 'CF ()', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache, must-revalidate', 'Set-Cookie': 'login_notice_check=%2F; path=/; secure; HttpOnly', 'Access-Control-Allow-Methods': 'GET,POST,OPTIONS,DELETE,PUT', 'Access-Control-Allow-Headers': 'DNT,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type', 'Access-Control-Max-Age': '1728000', 'Vary': 'Origin', 'Content-Encoding': 'gzip'} ResponseBody: []

### Code

```
  withdrawals = exchange.fetch_withdrawals()

```

**Comment by ttodua** [comment]
U

**Comment by ttodua** [comment]
I will try to add `fetchTransfers` implementation today and once this issue becomes automatically-closed, it means PR becomes merged, and then you can update ccxt package and `fetchTransfers` will be available for gate exchange

**Comment by dwiltech** [comment]
Yes the fetchTransfers would be good, however, I really am actually trying to get the withdrawals. Probably, I realize now, I did not need to include the fact the the funds were transferred from a Portfolio Margin Account into what Gate.io calls an "Exchange Account". It is from the "Exchange Account" that withdrawals can be made. However, my fetch_withdrawals() (which I would think would query against the "Exchange Account") comes up empty. Do I need to add some parameter to the fetch?

**Comment by dwiltech** [comment]
I discovered that gateio requires a different set of credentials for the spot account than for the Portfolio Margin Account. Using those spot credentials I was able to successfully use fetch_withdrawals() to get the recent withdrawals

**Comment by kroitor** [comment]
Hello @dwiltech, closing this during a cleanup — your own follow up in the thread turned out to be the resolution: gateio issues separate credentials per account type, and the portfolio margin account's withdrawals are reachable with the spot account credentials, as you confirmed working. Nothing to change in the library for this one. Thanks for reporting back with the solution, that is what made this closeable!
